### PR TITLE
OdkFormTest

### PR DESCRIPTION
**Pull request by joeflack4**
Improved maintainability of OdkFormTest by...
- (1) reducing size of test file, 
- (2) refactoring, and
- (3) Updated OdkFormTest to use a more appropriate assert method.

### CODE REVIEW

File: test/test_ppp.py [hunk 1]
```diff
@@ -193,65 +193,45 @@ class OdkFormTest(unittest.TestCase, PppTest):
...
-            ({'file': 'FQ-nut.xlsx', 'position': 25},
-             {'class': OdkGroup,
-              'repr': '<OdkGroup FQA: [<OdkPrompt age_warn>, <OdkPrompt '
-                      'age_diff>, <OdkPrompt age_same>, <OdkPrompt age>]>'}),
-            ({'file': 'FQ-nut.xlsx', 'position': 50},
-             {'class': OdkPrompt, 'repr': '<OdkPrompt nb_age_youngest>'}),
-            ({'file': 'FQ-nut.xlsx', 'position': 75},
-             {'class': OdkPrompt, 'repr': '<OdkPrompt ac_bp_check>'}),
-            ({'file': 'FQ-nut.xlsx', 'position': 100},
-             {'class': OdkPrompt, 'repr': '<OdkPrompt over_2yr_warning>'}),
```
Comment (joeflack4) [inline test/test_ppp.py]: Changed data structure of expected inputs / outputs.

File: test/test_ppp.py [hunk 2]
```diff
@@ -180,78 +180,48 @@ def test_render_header(self):
...
+            {'inputs': {'file': 'OdkFormTest.xlsx'}, 'position': 0,
```
Comment (joeflack4) [inline test/test_ppp.py]: Changed data structure of expected inputs/outputs.

File: test/test_ppp.py [hunk 3]
```diff
@@ -180,78 +180,48 @@ def test_render_header(self):
...
 
     def test_questionnaire(self):
         """Test expected results of converted questionnaire based on position.
         """
         forms = self.get_forms(self.data)
-        for i, expected_output in self.data:
-            output = forms[i['file']].questionnaire[i['position']]
-            self.assertTrue(str(output) == expected_output['repr'])
-            self.assertTrue(isinstance(output, expected_output['class']))
+
```
Comment (joeflack4) [inline test/test_ppp.py]: Added
- Error messages

Refactors in line with updated input/output data structure.

File: test/test_ppp.py [hunk 4]
```diff
@@ -180,78 +180,48 @@ def test_render_header(self):
...
+        for datum in self.data:
+            expected_output = datum['outputs']
+            output = \
+                forms[datum['inputs']['file']].questionnaire[datum['position']]
+
+            # - Check Object Representation
+            got = str(output)
+            expected = expected_output['repr']
+            msg = '\nGot: {}\nExpected: {}'.format(got, expected)
+            self.assertTrue(got == expected, msg)
```
Comment (jkpr) [inline test/test_ppp.py]: Improvement is `self.assertEqual(got, expected, msg)`. Please make the change.

**Comment by jkpr** [comment]
Please combine into one commit.

File: test/test_ppp.py (hunk 4 above)
Comment (joeflack4) [reply test/test_ppp.py]: Squashed and done~

**Comment by joeflack4** [comment]
@jkpr 
> Please combine into one commit.  

Squashed and done~

**Comment by jkpr** [comment]
@joeflack4 this is still showing as two commits.

**Comment by joeflack4** [comment]
@jkpr Squashed again~. Just one commit now.

**Pull request merged**